modules/container/src/main/java/org/mule/runtime/container/api/MuleInvocationHandler.java

```diff
@@ -21,35 +21,36 @@
 import java.util.List;
 
 /**
- * Extends {@link InvocationHandler} to provide and expose metadata about the inner {@link Service} implementation.
+ * Extends {@link InvocationHandler} to provide and expose metadata about the inner object implementation.
  * <p>
- * This allows for nested {@link Service} {@link Proxy}ies to work as expected.
+ * This allows for nested object {@link Proxy}ies to work as expected.
  * 
  * @since 4.0
  */
-public abstract class ServiceInvocationHandler implements InvocationHandler {
+public abstract class MuleInvocationHandler<T> implements InvocationHandler {
```

Comment:
Can you find a name that is more descriptive than MuleInvocationHandler based on what this class does?

Reply:
I agree with you, had trouble choosing one as _InvocationHandler_ was already taken by the interface, so I took the lazy road... As it adds a reflective capabilities (reading all methods from the inner object), what do you think about *ReflectiveInvocationHandler*? If you come up with anything better please let me know!

Reply:
"Reflective"  is redeundant as every invocation handler uses reflection. Based on the class javadoc, I would use MetadataInvocationHandler or something similar